src/zope/traversing/namespace.py

```diff
@@ -176,7 +221,7 @@ def queryResource(site, name, request, default=None):
     # resource to do this.  We still return the proxied resource.
     r = removeSecurityProxy(resource)
 
-    r.__parent__ = site
+    r.__parent__ = context
```

Comment:
I'm not convinced by this renaming.  As far as I remember, in Zope 3 resources are only directly available on sites, not on arbitrary traversal contexts.

Reply:
From configure.zcml, that namespace appears to be available for any type of object:
```xml
<adapter
    name="resource"
    for="* zope.publisher.interfaces.IRequest"
    provides="zope.traversing.interfaces.ITraversable"
    factory="zope.traversing.namespace.resource"
    />
```

Reply:
One place `++resource` is used turns out to be in [zope.viewlet](http://zopeviewlet.readthedocs.io/en/rtd-docs/viewlet.html) for CSS and JavaScript links. IIUC, this namespace only looks at the request object because it has to do with skins/layers.

Reply:
I may be conflating things.  The way I remember things work, you use something like

    <link rel="stylesheet" tal:attributes="href context/++resource++bar.css" />

in a template, and it'll render to a URL like `http://localhost:8080/@@/bar.css`.  The nameless view for actually serving resources is registered only for ISite.

I'm currently baffled by which part of this chain is making sure the `__parent__` of the resource is actually a site... Oh!  There's a custom IAbsoluteURL implementation for resources that uses `getSite()` instead of the resource's own `__parent__` that does this.

Ok, your change is correct and good.  Thank you!